### ISSUE TITLE
exec: don't expose "unexpected error from the vectorized runtime"

### ISSUE THREAD

**Issue by solongordon** [cockroachdb/cockroach#38944]
When vectorized execution panics we currently bubble up an "unexpected error from the vectorized runtime" to the user. This causes `TestNonRetryableError` to fail due to the error not being what it expects. I don't think this error makes sense to expose anything about the "vectorized runtime" to users anyway, so probably better to log this and bubble up the original error.

**Comment by yuzefovich** [comment]
I think we want to relax the test here. My thinking is that it is an artificial scenario, and in real usage the user shouldn't be able to see the "unexpected error ..." (unless it was really unexpected) because the error should contain a PG code (to the best of my understanding).

https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/exec/error.go#L52-L57
```
// Any error without a code already is "surprising" and
// needs to be annotated to indicate that it was
// unexpected.
if code := pgerror.GetPGCode(e); code == pgcode.Uncategorized {
	e = errors.Wrap(e, "unexpected error from the vectorized runtime")
}
```

**Comment by yuzefovich** [comment]
This has been fixed by #38990.
